ms/MSOper/MSMetaData.cc

```diff
@@ -3878,17 +3878,23 @@ std::pair<MDirection, MDirection> MSMetaData::getPointingDirection(
 		row >= this->nRows(),
 		"Row number exceeds number of rows in the MS"
 	);
-	SHARED_PTR<Vector<Int> > ant1, ant2;
-	_getAntennas(ant1, ant2);
-	antenna1 = (*ant1)[row];
-	antenna2 = (*ant2)[row];
-	time = (*_getTimes())[row];
+	// KS note: 2015/10/30
+	// getColum method (also used in _getAntennas and _getTimes) is
+	// too expesive for the current purpose of getting single cell.
+	String ant1ColName = MeasurementSet::columnName(MSMainEnums::ANTENNA1);
+	String ant2ColName = MeasurementSet::columnName(MSMainEnums::ANTENNA2);
+	antenna1 = ROScalarColumn<Int>(*_ms, ant1ColName).get(row);
+	antenna2 = ROScalarColumn<Int>(*_ms, ant2ColName).get(row);
+	bool autocorr = (antenna1==antenna2);
+	String timeColName = MeasurementSet::columnName(MSMainEnums::TIME);
+	time = ScalarColumn<Double>(*_ms, timeColName).get(row);
 	ROMSPointingColumns pCols(_ms->pointing());
 	Int pidx1, pidx2;
 	pidx1 = pCols.pointingIndex(antenna1, time, initialguess);
-	pidx2 = pCols.pointingIndex(antenna2, time, initialguess);
+	if (autocorr) pidx2 = pidx1;
```

Comment:
one line if is inconsistent with the rest of the file which uses
    if {
      //
    }